readthedocs/search/views.py

```diff
@@ -126,15 +129,13 @@ def elastic_search(request, project_slug=None):
                         ),
                     }
                     for hit in sorted(all_results, key=utils._get_hit_score, reverse=True)
-                ]
+                )
 
-                hit['inner_hits'].pop('sections', None)
-                hit['inner_hits'].pop('domains', None)
-                hit['inner_hits'] = sorted_results
+                result.meta.inner_hits = sorted_results
 
-        except:
+        except Exception as e:
             # if the control comes in this block,
-            # that implies that there was PageSearch
+            # that implies that there was a PageSearch
             pass
```

Comment:
We should log this `log.exception` instead of `pass`, or fix this logic. We shouldn't hit this in a normal case.

Reply:
I have changed the logic.
Turns out that a simple `if` was sufficient. :smiley: 
I have still wrapped the code in `try... except` block to log any exceptions (if occur) to help in debugging later.